test/integration/scheduler/framework_test.go

```diff
@@ -1022,22 +1027,39 @@ func TestUnreservePlugin(t *testing.T) {
 
 			if test.preBindFail {
 				if err = wait.Poll(10*time.Millisecond, 30*time.Second, podSchedulingError(testCtx.ClientSet, pod.Namespace, pod.Name)); err != nil {
-					t.Errorf("Expected a scheduling error, but didn't get it. error: %v", err)
+					t.Errorf("Expected a scheduling error, but didn't get it: %v", err)
 				}
-				if unreservePlugin.numUnreserveCalled == 0 || unreservePlugin.numUnreserveCalled != preBindPlugin.numPreBindCalled {
-					t.Errorf("Expected the unreserve plugin to be called %d times, was called %d times.", preBindPlugin.numPreBindCalled, unreservePlugin.numUnreserveCalled)
+				for i, pl := range reservePlugins {
+					if pl.numUnreserveCalled != 1 {
```

Comment:
nit: we don't need this counter, given that you are checking the channel below.

Reply:
it still helps to check that the number of Unreserve calls isn't greater than one (the channel could have multiple buffered events from the same plugin)

Reply:
actually, nvm, you're right we don't need this check given the idempotency guarantee